CMakeLists.txt

```diff
@@ -103,6 +103,14 @@ endif()
 # provide a namespaced alias for clients to 'link' against if catch is included as a sub-project
 add_library(Catch2::Catch2 ALIAS Catch2)
 
+if (NOT NOT_SUBPROJECT)
+    add_library(Catch2Main src/catch_with_main.cpp)
+
+    target_link_libraries(Catch2Main Catch2)
+
+    add_library(Catch2::Main ALIAS Catch2Main)
+endif ()
```

Comment:
Couple of things here:

* I am unclear on why this is enabled only if Catch2 is a subproject...
* The alias needs to be `Catch2::Catch2Main`*, because the alias library exists to make usage of Catch2-as-a-subproject and Catch2-as-an-installed-library uniform. During installation, the targets are prefixed with "namespace" `Catch2::`, so the alias prepends that "namespace" to real targets.
* Again, I would prefer `Catch2WithMain` as the target name.



\* Actually, since this is available only if Catch2 is a subproject, the alias doesn't have to be there at all.

Reply:
Fixed the naming, the subproject thing was there before. No idea why or how this is required.

Reply:
The reason is to avoid intruding upon the parent project's build with our tests, and its install step... If you don't have an explicit reason to do this, I'll clean it up and merge it afterwards.

Reply:
No reason from my side 👍